packages/baby-jubjub/src/baby-jubjub.ts

```diff
@@ -75,24 +76,14 @@ export function addPoint(p1: Point<bigint>, p2: Point<bigint>): Point<bigint> {
  * @returns The resulting point representing the public key.
  */
 export function mulPointEscalar(base: Point<bigint>, e: bigint): Point<bigint> {
-    if (scalar.isZero(e)) {
-        return [BigInt(0), BigInt(1)]
-    }
-    const eBits: Array<boolean> = []
-    while (!scalar.isZero(e)) {
-        if (scalar.isOdd(e)) {
-            eBits.push(true)
-        } else {
-            eBits.push(false)
-        }
-        e = scalar.shiftRight(e, BigInt(1))
-    }
+    e %= order
```

Comment:
Good thought.  I found myself wondering whether this was needed too, and whether it should be a modular reduction or an exception thrown for overly-large values.

Reply:
I think it is better to apply the reduction unless it implies another timing vulnerability, which is something I'm not sure about.

Reply:
```suggestion
    e %= order
    e += 1n << 254n
```

I suggest this so that the operation on line 88 will always operate on an `e` value of the same size. I'm not sure if the modular reduction is variable time (i'd guess yes).

Reply:
Am I understanding right that your proposal is to set the 254th bit to ensure uniform input size, and then intentionally ignore that bit in the for loop below?  If so, I can understand it, though it's very tricky and deserves a comment to explain it to future developers.

Reply:
A better number library, hopefully faster, would certainly be welcome, but I don't know how feasible it is.
I think there's a strategic decision to be made here which I'm not the right person to make.  How much is it worth trading speed and complexity against a partial solution to timing attacks.  There might be more full solutions to both timing and performance, but how difficult are they, and how likely are they to be accomplished soon?

Reply:
Yes, i'm pushing for PSE to begin maintaining an open source number library with constant time operations written in Rust, but supporting webassembly. I mostly want this library for building post quantum secure constructions, but i think we could add support for things like this. Ideally we pull all this logic out of JS and into wasm where we can be much more certain of what is executing, and hopefully get equal or better performance.

This is a next couple of months type goal though, certainly not an immediate fix. @cedoor I think you're going to be the authority on this PR. I personally don't think it's worth significantly degrading performance for a fix that we're not sure solves the problem.